### PR TITLE
Migrate GCM to FCM

### PR DESCRIPTION
**Pull request by enahum**
After this is merged the push proxy will start using FCM instead of GCM.

GCM is being taken down on April 11.

Once this PR is merged we need to update the `AndroidApiKey` to use the new format, the current one will still work but is better if we update it.

image

### CODE REVIEW

File: server/android_notification_server.go [hunk 1]
```diff
@@ -62,28 +62,30 @@ func (me *AndroidNotificationServer) SendNotification(msg *PushNotification) Pus
 		}
 	}
 
-	regIDs := []string{msg.DeviceId}
-	gcmMsg := gcm.NewMessage(data, regIDs...)
-
-	sender := &gcm.Sender{
-		ApiKey: me.AndroidPushSettings.AndroidApiKey,
-		Http:   httpClient,
```
Comment (hmhealey) [inline server/android_notification_server.go]: Do we need to pass the http client to the new library somehow to use the timeouts and such we have specified [here](https://github.com/mattermost/mattermost-push-proxy/blob/4fcc302a9dd0d1fc5449c77550c5d4cb79ceef68/server/push_notification.go#L30)?

Comment (enahum) [reply server/android_notification_server.go]: Not really it uses it internally, I think we can can remove it, I’ll check if its needed for ios

Comment (enahum) [reply server/android_notification_server.go]: @hmhealey removed

Comment (hmhealey) [reply server/android_notification_server.go]: Does the built-in one have the correct timeouts and such though? I assume there was a reason we specific our own client as opposed to just using the default one from `net/http`

Comment (enahum) [reply server/android_notification_server.go]: you can take a look at the library.. the previous one required an httpClient, this one does not

Comment (enahum) [reply server/android_notification_server.go]: Taken from the library

```go
const (
	// DefaultEndpoint contains endpoint URL of FCM service.
	DefaultEndpoint = "https://fcm.googleapis.com/fcm/send"

	// DefaultTimeout duration in second
	DefaultTimeout time.Duration = 30 * time.Second
)
```

Comment (hmhealey) [reply server/android_notification_server.go]: Ah okay. Makes sense then

**Comment by enahum** [comment]
@cpanato can we build a release for this one and also can we publish it in HPNS and TPNS.

Btw look me up cause I need to give you the new ApiKey

**Comment by enahum** [comment]
@cpanato do you need a ticket to go update this?

**Comment by cpanato** [comment]
@enahum missed the previous message. I will update. but I would like to tag this should we tag as `v5.8.0` ?

**Pull request merged**